src/app/application.cpp

```diff
@@ -403,15 +403,63 @@ void Application::processParams(const QStringList &params)
         return;
     }
 #endif
+    BitTorrent::AddTorrentParams torrentParams;
+    bool skipTorrentDialog = false;
 
     foreach (QString param, params) {
         param = param.trimmed();
+
+        // Process strings indicating options specified by the user. Use of
+        // any of these should prevent the "Add new torrent" dialog from
+        // opening:
+
+        if (param.startsWith("@savePath=")) {
+            torrentParams.savePath = param.mid(10);
+            skipTorrentDialog = true;
```

Comment:
Actually this may require some future proofing. If someone were to make the change we're discussing, it would change the behavior of these command line arguments from preventing the "Add new torrent" dialog from opening to not preventing it. I think it's bad form to have a potential future version of the app change the behavior of command line arguments in that dramatic of a fashion.

So how about this: I'll add the TODO and change the `--skip-dialog` to be a tri-state bool. This already makes sense because there is actually an application-wide setting for whether the add torrent dialog should be opened, so being able to say `--skip-dialog=false` might be useful in some cases. For now I'll make a note in the command line usage text that using `--skip-dialog=false` in conjunction with the other command line arguments I've added is currently unsupported.

But that way if in a future version someone follows up on the TODO and changes `AddNewTorrentDialog` so that you can override its settings with these command line arguments, we can start allowing using them in conjunction with `--skip-dialog=false` to allow someone to pop open the dialog, override its settings, all while not change the existing behavior of the other command line arguments. Sound good?